spec/classes/puppet_agent_configure_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'puppet_agent::configure' do
+  let(:pre_condition) do
+    [ 'function assert_private() { }',
+      'class puppet_agent::params { $config = "/etc/puppetlabs/puppet/puppet.conf" }',
+      'include puppet_agent::params',
+      'include puppet_agent',
+    ]
+  end
+
+  context 'with empty config input' do
+    let(:pre_condition) do
+      [ super(),
+        'class puppet_agent { $config = [ ] }',
+      ]
+    end
+
+    it { is_expected.to compile }
+  end
+
+  context 'with settings input' do
```

Comment:
Should we have some tests that assert the settings' absence when `ensure => absent` (regardless of whether  `value` is set)?